src/android.ts

```diff
@@ -237,6 +238,121 @@ export class AndroidRobot implements Robot {
 
 		return rotation === "0" ? "portrait" : "landscape";
 	}
+
+	private getParsedXml(): UiAutomatorXml {
+		const dump = this.adb("exec-out", "uiautomator", "dump", "/dev/tty");
+
+		const parser = new xml.XMLParser({
+			ignoreAttributes: false,
+			attributeNamePrefix: ""
+		});
+
+		return parser.parse(dump) as UiAutomatorXml;
+	}
+
+	private getScreenElementRect(node: UiAutomatorXmlNode): ScreenElementRect {
+		const bounds = String(node.bounds);
+
+		const [, left, top, right, bottom] = bounds.match(/^\[(\d+),(\d+)\]\[(\d+),(\d+)\]$/)?.map(Number) || [];
+		return {
+			x: left,
+			y: top,
+			width: right - left,
+			height: bottom - top,
+		};
+	}
+
+	// --- TV Specific Methods ---
+
+	private async tvTap(x: number, y: number): Promise<void> {
+		let currentDirection = this.getDpadDirection(x, y);
+
+		while (currentDirection) {
+			this.pressDpad(currentDirection);
+			currentDirection = this.getDpadDirection(x, y);
+		}
+
+		// Upon breaking from the while loop, we have reached at the target coordinates
+		// Press the dpad center button to 'click' on the focused element
+		this.pressDpad("DPAD_CENTER");
+	}
```

Comment:
`memo`
self explanatory.
- based on the current focused item and the target x, y coordinates, I device which D-pad direction I need to press.
- if I've reached the destination, `getDpadDirection` returns null which is when I perform the TV version of click i.e., D-pad Center button